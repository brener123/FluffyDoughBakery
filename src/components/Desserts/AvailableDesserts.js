//import { useEffect, useState } from 'react';

import Card from '../UI/Card';
import DessertItem from './DessertItem/DessertItem';
import classes from './AvailableDesserts.module.css';


const DUMMY_DESSERTS = [
  {
    id: 'm1',
    name: 'Apple & Blackberry Crumble',
    description: 'Best pie you ever tried!',
    price: 22.99,
  },
  {
    id: 'm2',
    name: 'Banana Pancake',
    description: 'Our specialty!',
    price: 16.5,
  },
  {
    id: 'm3',
    name: 'Choc Chip Pecan Pie',
    description: 'Part of our rotating desserts and a deal of the day!',
    price: 12.99,
  },
  {
    id: 'm4',
    name: 'Chocolate Caramel Crispy cake',
    description: 'Melts in your mouth!',
    price: 18.99,
  },
];

const AvailableDesserts = () => {
  const DessertsList = DUMMY_DESSERTS.map((Dessert) => (
    <DessertItem
      key={Dessert.id}
      id={Dessert.id}
      name={Dessert.name}
      description={Dessert.description}
      price={Dessert.price}
    />
  ));

  return (
    <section className={classes.desserts}>
      <Card>
        <ul>{DessertsList}</ul>
      </Card>
    </section>
  );
};

export default AvailableDesserts;
