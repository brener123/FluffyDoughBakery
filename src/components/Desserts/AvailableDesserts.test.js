import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import AvailableDesserts from './AvailableDesserts';

test('renders Available Desserts and get the Banana pancake element', () => {
  render(<AvailableDesserts />);
  const dessertElement = screen.getByText('Banana pancake', { exact: false });
  expect(dessertElement).toBeInTheDocument();
});

test('renders correctly by snapshot', () => {
  const tree = renderer
    .create(<AvailableDesserts/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
