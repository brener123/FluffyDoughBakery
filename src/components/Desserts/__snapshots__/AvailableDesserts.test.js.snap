// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly by snapshot 1`] = `
<section
  className="desserts"
>
  <div
    className="card"
  >
    <ul>
      <li
        className="dessert"
      >
        <div>
          <h3>
            Apple & Blackberry Crumble
          </h3>
          <div
            className="description"
          >
            Best pia you ever tried!
          </div>
          <div
            className="price"
          >
            $22.99
          </div>
        </div>
        <div>
          <form
            className="form"
            onSubmit={[Function]}
          >
            <div
              className="input"
            >
              <label
                htmlFor="amount"
              >
                Amount
              </label>
              <input
                defaultValue="1"
                id="amount"
                max="5"
                min="1"
                step="1"
                type="number"
              />
            </div>
            <button>
              + Add
            </button>
          </form>
        </div>
      </li>
      <li
        className="dessert"
      >
        <div>
          <h3>
            Banana Pancake
          </h3>
          <div
            className="description"
          >
            Our specialty!
          </div>
          <div
            className="price"
          >
            $16.50
          </div>
        </div>
        <div>
          <form
            className="form"
            onSubmit={[Function]}
          >
            <div
              className="input"
            >
              <label
                htmlFor="amount"
              >
                Amount
              </label>
              <input
                defaultValue="1"
                id="amount"
                max="5"
                min="1"
                step="1"
                type="number"
              />
            </div>
            <button>
              + Add
            </button>
          </form>
        </div>
      </li>
      <li
        className="dessert"
      >
        <div>
          <h3>
            Choc Chip Pecan Pie
          </h3>
          <div
            className="description"
          >
            Part of our rotating desserts and a deal of the day!
          </div>
          <div
            className="price"
          >
            $12.99
          </div>
        </div>
        <div>
          <form
            className="form"
            onSubmit={[Function]}
          >
            <div
              className="input"
            >
              <label
                htmlFor="amount"
              >
                Amount
              </label>
              <input
                defaultValue="1"
                id="amount"
                max="5"
                min="1"
                step="1"
                type="number"
              />
            </div>
            <button>
              + Add
            </button>
          </form>
        </div>
      </li>
      <li
        className="dessert"
      >
        <div>
          <h3>
            Chocolate Caramel Crispy cake
          </h3>
          <div
            className="description"
          >
            Melts is your mouth!
          </div>
          <div
            className="price"
          >
            $18.99
          </div>
        </div>
        <div>
          <form
            className="form"
            onSubmit={[Function]}
          >
            <div
              className="input"
            >
              <label
                htmlFor="amount"
              >
                Amount
              </label>
              <input
                defaultValue="1"
                id="amount"
                max="5"
                min="1"
                step="1"
                type="number"
              />
            </div>
            <button>
              + Add
            </button>
          </form>
        </div>
      </li>
    </ul>
  </div>
</section>
`;
